### PR TITLE
YouView payload hashing to avoid sending unchanged items

### PR DESCRIPTION
**Pull request by emilssolmanis**

### CODE REVIEW

File: src/main/java/org/atlasapi/feeds/tasks/youview/creation/TaskCreationTask.java [hunk 1]
```diff
@@ -56,6 +69,8 @@ public TaskCreationTask(YouViewLastUpdatedStore lastUpdatedStore,
         this.taskStore = checkNotNull(taskStore);
         this.taskCreator = checkNotNull(taskCreator);
         this.payloadCreator = checkNotNull(payloadCreator);
+        this.payloadHashStore = payloadHashStore;
```
Comment (tfm) [inline src/main/java/org/atlasapi/feeds/tasks/youview/creation/TaskCreationTask.java]: *cough* `checkNotNull` *cough*

File: src/main/java/org/atlasapi/feeds/youview/persistence/MongoYouViewPayloadHashStore.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    public void saveHash(HashType payloadType, String imi, String hash) {
+        DBObject dbObject = BasicDBObjectBuilder
+                .start(MongoConstants.ID, key(payloadType, imi))
+                .add(HASH_FIELD, hash)
+                .get();
+        collection.save(dbObject);
+    }
+
+    @Override
+    public Optional<String> getHash(HashType payloadType, String imi) {
```
Comment (tfm) [inline src/main/java/org/atlasapi/feeds/youview/persistence/MongoYouViewPayloadHashStore.java]: This isn't necessarily an imi, is it? it's the object key.

**Comment by tfm** [comment]
:+1: save for a minor naming thing

**Comment by Dalamar42** [comment]
LGTM

**Comment by emilssolmanis** [comment]
@tfm I saw you make a rebase branch or something in mail today, assuming this can be closed then?

**Comment by emilssolmanis** [comment]
Closed since this was merged another way.

**Pull request closed without merging**